### PR TITLE
improve error message on int, float with 1 dimension.

### PR DESCRIPTION
**Pull request by redbopo**
fix [issues-698](https://github.com/lcompilers/lpython/issues/698).
Add with function `std::string type_1dim_helper()`

### CODE REVIEW

**Comment by redbopo** [comment]
Thanks @certik , now fix tests.
This patch also influcenes the  `integration_tests/vec_01.py`
with  `tests/reference/pass_loop_vectorise-vec_01-be9985e.stdout` 
And in stdout format like
`vector_copy_f64f64i32 @IntrinsicOptimization` 
will become 
` vector_copy_f64[9216] f64[9216] i32   @IntrinsicOptimization` .

File: src/libasr/asr_utils.h [hunk 1]
```diff
@@ -195,26 +195,39 @@ static inline std::string type_to_str(const ASR::ttype_t *t)
...
             ASR::Real_t *r = (ASR::Real_t*)t;
+            std::string res = "";
             switch (r->m_kind) {
-                case 4: { return "f32"; }
-                case 8: { return "f64"; }
+                case 4: { res = "f32"; break; }
+                case 8: { res = "f64"; break; }
                 default: { throw LFortranException("Float kind not supported"); }
             }
+            if (r->n_dims == 1) res = type_1dim_helper(res, r->m_dims->m_end);
```
Comment (czgdp1807) [inline src/libasr/asr_utils.h:229]: This is okay. But may be we can generalise to `n` dimensions. Once we support multi-dimensions then we will update this.

**Comment by czgdp1807** [comment]
> vector_copy_f64[9216] f64[9216] i32 @IntrinsicOptimization

Ah! wait this ain't looking good.

File: src/libasr/asr_utils.h [hunk 2]
```diff
@@ -195,26 +195,39 @@ static inline std::string type_to_str(const ASR::ttype_t *t)
...
-                case 2: { return "i16"; }
-                case 4: { return "i32"; }
-                case 8: { return "i64"; }
+                case 1: { res = "i8"; break; }
+                case 2: { res = "i16"; break; }
+                case 4: { res = "i32"; break; }
+                case 8: { res = "i64"; break; }
                 default: { throw LFortranException("Integer kind not supported"); }
             }
+            if (i->n_dims == 1) res = type_1dim_helper(res, i->m_dims->m_end);
```
Comment (czgdp1807) [inline src/libasr/asr_utils.h:218]: And update this as  follows,

```suggestion
            if (i->n_dims == 1 && for_error_message) res = type_1dim_helper(res, i->m_dims->m_end);
```

**Review by czgdp1807 — CHANGES_REQUESTED** [review]
https://github.com/lcompilers/lpython/pull/708#discussion_r909824337 and https://github.com/lcompilers/lpython/pull/708#discussion_r909827390

**Review by czgdp1807 — APPROVED** [review]

**Review by namannimmo10 — APPROVED** [review]

**Comment by namannimmo10** [comment]
Thanks!

**Pull request merged**